sympy/physics/units/quantities.py

```diff
@@ -63,13 +64,19 @@ def __new__(cls, name, abbrev=None, dimension=None, scale_factor=None,
         elif isinstance(abbrev, str):
             abbrev = Symbol(abbrev)
 
+        # HACK: These are here purely for type checking. They actually get assigned below.
+        cls._is_prefixed = is_prefixed
+        cls._is_physics_constant = is_physics_constant
+
         obj = AtomicExpr.__new__(cls, name, abbrev)
         obj._name = name
         obj._abbrev = abbrev
         obj._latex_repr = latex_repr
         obj._unicode_repr = pretty_unicode_repr
         obj._ascii_repr = pretty_ascii_repr
         obj._mathml_repr = mathml_presentation_repr
+        obj._is_prefixed = is_prefixed
+        obj._is_physics_constant = is_physics_constant
```

Comment:
Be careful that this information is lost in case you call `Quantity(*quantity_variable.args)` or `Quantity.func(*quantity_variable.args)`, because `is_prefixed` and `is_physics_constant` are not passed to the superclass as arguments.

This could be an issue if you call some SymPy functionality like the term rewriter or the substitution.

Reply:
What about subclassing `Quantity` into a new class `Constant` just defining `is_constant` as a static field?
e.g.
```python
class Quantity:
    is_physics_constant = False
    ...

class Constant(Quantity):
    is_physics_constant = True
```

Reply:
> Be careful that this information is lost

Could you give me a test case for this? I'm not familiar enough with this SymPy behavior.

> What about subclassing `Quantity`

That seems like a better idea.